### ISSUE TITLE
missing throughput causes a deserialization error

### ISSUE THREAD

**Issue by lu-zero** [BurntSushi/critcmp#1]
Currently `critcmp` from *crates.io* seems to have problems parsing the content of the currently released `criterion`.

```
/home/lu_zero/rav1e/target/criterion/get_sad/BLOCK_64X128/new/estimates.json: invalid type: null, expected a string at line 1 column 40
```
The file:
```
{"Mean":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.33967490751667,"upper_bound":147.43358882620052},"point_estimate":147.3849476164319,"standard_error":0.023965715727952808},"Median":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.30833994178107,"upper_bound":147.4066363541041},"point_estimate":147.34522830111064,"standard_error":0.025997038369799854},"MedianAbsDev":{"confidence_interval":{"confidence_level":0.95,"lower_bound":0.1509735509193759,"upper_bound":0.24146680313029364},"point_estimate":0.1924932858498119,"standard_error":0.024073251188416186},"Slope":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.31067985782533,"upper_bound":147.3925789529354},"point_estimate":147.35072958201718,"standard_error":0.020952924148138635},"StdDev":{"confidence_interval":{"confidence_level":0.95,"lower_bound":0.17501272137980164,"upper_bound":0.3100491684607925},"point_estimate":0.24054982719442242,"standard_error":0.035463059217927154}}
```

**Comment by BurntSushi** [comment]
critcmp supports the latest version of Criterion. That's not the issue here, but I did add that verbiage to the README.

The problem here is that critcmp wasn't handling the absence of throughput correctly. That's now fixed and `0.1.1` is on crates.io with the fix.

**Comment by lu-zero** [comment]
This still fails though:
```
{"Mean":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.33967490751667,"upper_bound":147.43358882620052},"point_estimate":147.3849476164319,"standard_error":0.023965715727952808},"Median":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.30833994178107,"upper_bound":147.4066363541041},"point_estimate":147.34522830111064,"standard_error":0.025997038369799854},"MedianAbsDev":{"confidence_interval":{"confidence_level":0.95,"lower_bound":0.1509735509193759,"upper_bound":0.24146680313029364},"point_estimate":0.1924932858498119,"standard_error":0.024073251188416186},"Slope":{"confidence_interval":{"confidence_level":0.95,"lower_bound":147.31067985782533,"upper_bound":147.3925789529354},"point_estimate":147.35072958201718,"standard_error":0.020952924148138635},"StdDev":{"confidence_interval":{"confidence_level":0.95,"lower_bound":0.17501272137980164,"upper_bound":0.3100491684607925},"point_estimate":0.24054982719442242,"standard_error":0.035463059217927154}}
```

**Comment by BurntSushi** [comment]
That's not helpful, sorry. That isn't the only file critcmp reads, and I don't know what you mean by "fail." Please provide your entire target/criterion directory in a tar archive.

Please consider that we are reading an undocumented internal format. There is no spec. Therefore, getting it right requires iteration. Please be patient and work with me while providing as much detail as possible. I can't guess at what you mean.

**Comment by lu-zero** [comment]
It is fine and I'd be glad to try to help, I'd ask for further direction on irc if you are available today.

The code that I'm using is in [this pr](https://github.com/xiph/rav1e/pull/610). I can try to reduce it since building the whole thing is not immediate (yet).

**Comment by BurntSushi** [comment]
@lu-zero Thanks! I'm not on IRC during work hours. I just need your `target/criterion` directory:

```
$ tar zcf lu-zero-target-criterion.tar.gz target/criterion
```

And then attach that to a github issue comment and I think that should be enough.

**Comment by lu-zero** [comment]
Apparently github does not like the file. I tried with zip and tar.gz 

You can find [here](https://dev.gentoo.org/~lu_zero/rav1e-criterion.tar.gz) the tarball anyway.

**Comment by BurntSushi** [comment]
I am getting a completely different error than what you reported:

```
$ critcmp
/tmp/critcmp-1/target/criterion/get_sad/BLOCK_128X64/base/estimates.json: No such file or directory (os error 2)
```

**Comment by lu-zero** [comment]
This is even stranger since the file is present that I could see.

**Comment by BurntSushi** [comment]
The error messages are bad here. It's trying to load `benchmark.json`, which is not present.

I tried running `critcmp` on the benchmarks produced by https://github.com/xiph/rav1e and I found a problem: `function_id` can be null. I fixed that and things seem to work.

I don't know why your tar archive doesn't include `benchmark.json`. If this is a real thing that can happen in Criterion, then we'll want to be more robust to that.

In the future, please point me to a repo and a series of commands I can run to get Criterion data.

**Comment by BurntSushi** [comment]
My latest bug fix here is in `critcmp 0.1.2` on crates.io.

**Comment by lu-zero** [comment]
Updating to the criterion 2.5.0 fixed the missing-`benchmark.json` problem.

Looks like it works now :) Thank you a lot!
